inference-engine/src/low_precision_transformations/include/low_precision/network_helper.hpp

```diff
@@ -270,6 +270,12 @@ template <typename T, typename... Args>
 std::shared_ptr<Node> fold_reshape(Args&&... args) {
     std::shared_ptr<Node> node = std::make_shared<T>(std::forward<Args>(args)...);
     if (node->get_output_size() == 1) {
+        // issue #57985: remove fold_reshape & reuse nGraph implementation
+        const auto values = as_type_ptr<opset1::Constant>(node->input_value(1).get_node_shared_ptr())->template cast_vector<int64_t>();
+        if (std::any_of(values.begin(), values.end(), [](const float value) { return (value == 0ul) || (value == -1); })) {
```

Comment:
If `values` vector is int64_t type, why `value` inside lambda is float?
Also 0 is unsigned long, but -1 is just int, actually strange decision.

Reply:
typo was fixed, thanks